build_helper.sh

```diff
@@ -65,6 +68,18 @@ else
   MVFST_INSTALL_DIR=$INSTALL_PREFIX
 fi
 
+# Default to parallel build width of 4.
+# If we have "nproc", use that to get a better value.
+# If not, try using sysctl to get a better value.
+set +x
+nproc=4
+if [ -z "$(hash nproc 2>&1)" ]; then
+    nproc=$(nproc)
+elif [ -z "$(hash sysctl 2>&1)" ]; then
+    nproc=$(sysctl -n hw.ncpu)
```

Comment:
Alternatively we can always hardcode it to 4 since Mac tends to freeze when using all CPUs.

Reply:
I don't have a strong opinion here, but isn't checking the number of CPUs slightly more defensive programming / resistant to being on a Mac with more than 4 cores?  (e.g., my home iMac has 8)

Reply:
Yeah. I just wanted to error on the side of slower build than freezing the users personal laptops as a default behavior :). We can of course make this an option user-supplied field as well.